modules/gnome-extensions/gnome-extensions.sh

```diff
@@ -3,88 +3,226 @@
+      # inform the user & fail the build
+      if [[ "${UUID}" == "null" ]]; then
+        echo "ERROR: Extension '${EXTENSION_NAME}' doesn't have 'uuid' key inside metadata.json"
+        echo "You may inform the extension developer about this error, as he can fix it"
+        exit 1
+      fi
+      if [[ "${EXT_GNOME_VER}" == "null" ]]; then
+        echo "ERROR: Extension '${EXTENSION_NAME}' doesn't have 'shell-version' key inside metadata.json"
+        echo "You may inform the extension developer about this error, as he can fix it"
+        exit 1
+      fi      
+      # Compare if extension is compatible with current Gnome version
+      # If extension is not compatible, inform the user & fail the build
+      if ! [[ "${EXT_GNOME_VER}" =~ "${GNOME_VER}" ]]; then
+        echo "ERROR: Extension '${EXTENSION_NAME}' is not compatible with current Gnome v${GNOME_VER}!"
+        exit 1
+      fi  
+      # Install main extension files
+      echo "Installing main extension files"
+      install -d -m 0755 "/usr/share/gnome-shell/extensions/${UUID}/"
+      find "${TMP_DIR}" -mindepth 1 -maxdepth 1 ! -path "*locale*" ! -path "*schemas*" -exec cp -r {} "/usr/share/gnome-shell/extensions/${UUID}/" \;
+      find "/usr/share/gnome-shell/extensions/${UUID}" -type d -exec chmod 0755 {} +
+      find "/usr/share/gnome-shell/extensions/${UUID}" -type f -exec chmod 0644 {} +
+      # Install schema
+      if [[ -d "${TMP_DIR}/schemas" ]]; then
+        echo "Installing schema extension file"
+        install -d -m 0755 "/usr/share/glib-2.0/schemas/"
+        install -D -p -m 0644 "${TMP_DIR}/schemas/"*.gschema.xml "/usr/share/glib-2.0/schemas/"
+      fi  
+      # Install languages
+      # Locale is not crucial for extensions to work, as they will fallback to gschema.xml
+      # Some of them might not have any locale at the moment
+      # So that's why I made a check for directory
+      if [[ -d "${TMP_DIR}/locale" ]]; then
+        echo "Installing language extension files"
+        install -d -m 0755 "/usr/share/locale/"
+        cp -r "${TMP_DIR}/locale"/* "/usr/share/locale/"
+      fi  
+      # Delete the temporary directory
+      echo "Cleaning up the temporary directory"
+      rm -r "${TMP_DIR}"
+      echo "Extension '${EXTENSION_NAME}' is successfully installed"
+      echo "----------------------------------INSTALLATION DONE----------------------------------"
+  done
+fi
+
+# New method of installing extensions
+if [[ ${#INSTALL[@]} -gt 0 ]] && ! "${LEGACY}"; then
+  for INSTALL_EXT in "${INSTALL[@]}"; do
+      # Replaces whitespaces with %20 for install entries which contain extension name, since URLs can't contain whitespace
+      WHITESPACE_HTML="${INSTALL_EXT// /%20}"
+      URL_QUERY=$(curl -s "https://extensions.gnome.org/extension-query/?search=${WHITESPACE_HTML}")
+      QUERIED_EXT=$(echo "${URL_QUERY}" | yq ".extensions[] | select(.name == \"${INSTALL_EXT}\")")
```

Comment:
```suggestion
      QUERIED_EXT=$(echo "${URL_QUERY}" | jq ".extensions[] | select(.name == \"${INSTALL_EXT}\")")
```

Use `jq` when parsing `json`.

Reply:
I thought to use `yq`, since we bundle it in the image.
But I can use `jq`, if it's seen as more reliable.

I replaced all `yq` instances with `jq`, since everything is JSON.

Reply:
Marking as unresolved, since I have to change the format of `jq` commands.

It also seems to print gathers double quotes, unlike `yq`, so I'll have to see how to get that fixed too.

Reply:
I fixed all `jq` commands now.

[`fd5e46f` (#238)](https://github.com/blue-build/modules/pull/238/commits/fd5e46fbe4422c15712d9ee858aea5a5bce43449)

Reply:
@fiftydinar AFAIK `jq` should support the familiar dot syntax too? Worth to test locally?

Reply:
> @fiftydinar AFAIK `jq` should support the familiar dot syntax too? Worth to test locally?

Tested it locally, familiar dot syntax works for most things, but for some reason, it fails for `gettext-domain` & `settings-schema`.

When using bracketed-like syntax, it works, so I also added it for other entries, just to be safe.